groupRowsBy - prop type error (again)

### Describe the bug

Unfortunately, there have been some issues in the past regarding the prop types of `groupRowsBy` (DataTable). Currently, there seem to be problems (again).
While it generally works to pass an array to groupRowsBy, which is handled perfectly in the browser, my IDE still reports the following:

```
error TS2322: Type 'string[]' is not assignable to type 'string | number | symbol | ((field: any) => object) | undefined'.

1305                 :groupRowsBy="['id', 'tag']"
                      ~~~~~~~~~~~

  node_modules/primevue/datatable/index.d.ts:1106:5
    1106     groupRowsBy?: ((field: NoInfer<T>) => object) | keyof T[] | keyof T | undefined;
             ~~~~~~~~~~~
    The expected type comes from property 'groupRowsBy' which is declared here on type 'DataTableProps<any> & VNodeProps & AllowedComponentProps & ComponentCustomProps & Record<string, unknown>'
```

Full example:

```
              <DataTable
                v-model:filters="filtersTags"
                :value="tagRows"
                rowGroupMode="rowspan"
                :groupRowsBy="['id', 'tag']"
                showGridlines
                :globalFilterFields="['tag', 'lang', 'text']"
              >
```

### Pull Request Link

_No response_

### Reason for not contributing a PR

- [ ] Lack of time
- [x] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Other Reason

_No response_

### Reproducer

-

### Environment

Not relevant

### Vue version

3.5.16

### PrimeVue version

4.3.5

### Node version

24.2.0

### Browser(s)

_No response_

### Steps to reproduce the behavior

Provide an array to groupRowsBy.

### Expected behavior

No prop type issue although it generally works.

--- a/packages/primevue/src/datatable/DataTable.d.ts
+++ b/packages/primevue/src/datatable/DataTable.d.ts
@@ -1103,7 +1103,7 @@ export interface DataTableProps<T = any> {
     /**
      * One or more field names to use in row grouping.
      */
-    groupRowsBy?: ((field: NoInfer<T>) => object) | keyof T[] | keyof T | undefined;
+    groupRowsBy?: ((field: NoInfer<T>) => object) | (keyof T)[] | keyof T | undefined;
     /**
      * Whether the row groups can be expandable.
      * @defaultValue false

